### PR TITLE
Add a hook that deploys Ceph on BMO nodes

### PR DESCRIPTION
**Pull request by pablintino**
This patch is to allow discovering of nodeset computes and adding them to a in memory inventory before calling ceph, this is to enable NFV+HCI job to run.

### CODE REVIEW

File: playbooks/ceph.yml
Comment (fmount) [inline playbooks/ceph.yml:66]: I assume this preserves the default behavior as the default value is what we previously had on L66.

File: hooks/playbooks/ceph-bm.yml
Comment (fmount) [inline hooks/playbooks/ceph-bm.yml:11]: is the $namespace something we should pass as a parameter? I saw this pattern somewhere and I'm wondering if it makes sense to have a `$namespace | default('openstack')`.

**Review by fmount — APPROVED** [review]
/lgtm

**Comment by fmount** [comment]
Just a small nit about the `namespace`, otherwise looks good to me. Not sure we have any job we can look at but I know @eshulman2 had positive results.

File: hooks/playbooks/ceph-bm.yml
Comment (cjeanner) [reply hooks/playbooks/ceph-bm.yml:11]: `namespace` is passed down by the run_hook role:
https://github.com/openstack-k8s-operators/ci-framework/blob/0f9117448b23647fa9051064c3e23cad87404651/roles/run_hook/tasks/playbook.yml#L35
You can/should consume it.

File: playbooks/ceph.yml
Comment (eshulman2) [reply playbooks/ceph.yml:66]: yes

Comment (cjeanner) [inline playbooks/ceph.yml:66]: care to split a bit that line? It's barely readable...
Something like:
```yaml
vars:
  _target_group: "{{ cifmw_ceph_target | default('computes') }}"
  _target: "{{ groups[_target] | default([]) | first }}"
 ansible_ssh_private_key_file: >-
  {{
    hostvars[_target]['ansible_ssh_private_key_file'] |
    default(lookup('env', 'ANSIBLE_SSH_PRIVATE_KEY'))
  }}
```

Also, I'm not sure the `| default([]) | first` works, iirc `first` will crash because there's no first element in an empty list... Care to validate that on the side?

File: playbooks/ceph.yml
Comment (cjeanner) [inline playbooks/ceph.yml:122]: same, please split that up to keep it readable...

File: hooks/playbooks/ceph-bm.yml
Comment (eshulman2) [reply hooks/playbooks/ceph-bm.yml:11]: It is possible but I don't think it is actually required having that I don't think anyone is using a different namespace for openstack other than openstack but lets hear some more opinions

**Review by cjeanner — CHANGES_REQUESTED** [review]

File: playbooks/ceph.yml
Comment (eshulman2) [reply playbooks/ceph.yml]: done

File: playbooks/ceph.yml
Comment (fultonj) [inline playbooks/ceph.yml:72]: these changes look like they will be fine but could we please test project them downstream to make sure they do not affect the unijobs?

File: hooks/playbooks/ceph-bm.yml
Comment (fultonj) [inline hooks/playbooks/ceph-bm.yml:11]: I see this is one command to generate a yaml file so it's probably fine.

Did you consider using the `kubernetes.core.k8s` module? For example.

https://github.com/openstack-k8s-operators/ci-framework/blob/main/roles/ci_lvms_storage/tasks/main.yml#L42-L52'

In general I think this is nicer than running oc commands but I don't think we have to block the review over it.

File: playbooks/ceph.yml
Comment (fultonj) [inline playbooks/ceph.yml:128]: ditto

**Review by fultonj — COMMENTED** [review]
Thank you. In general I'm positive on this. Can we make sure it tested downstream before merging? If you confirm I could approve if you like.

File: playbooks/ceph.yml
Comment (eshulman2) [reply playbooks/ceph.yml:72]: yes I am testing it now

File: hooks/playbooks/ceph-bm.yml
Comment (eshulman2) [reply hooks/playbooks/ceph-bm.yml:11]: I'll look into it but I would prefer doing it after getting this patch merge having that those are leftovers we wanted to get done ASAP.

**Comment by eshulman2** [comment]
recheck

**Review by fultonj — APPROVED** [review]
/approve
/lgtm

**Pull request merged**
